src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -122,6 +122,159 @@ void LambertSolver::problemGeometry()
     this->O_frame2 = {i_r2, i_t2, i_h};
 }
 
+/*! This method computes the initial guess for the free variable x
+    @param lam lambda parameter that defines the problem geometry
+    @param T non-dimensional time-of-flight
+    @return std::vector<double>
+*/
+std::vector<double> LambertSolver::getInitialGuess(double lam, double T)
+{
+    // T for x=0 and zero revolution
+    double T00 = acos(lam) + lam*sqrt(1.0-pow(lam, 2));
+    // T forx=1 (parabolic case)
+    double T1 = 2.0/3.0*(1.0 - pow(lam, 3));
+    // T for x=0 and M revolutions
+    double T0M = T00 + this->M*M_PI;
+
+    double x0_sol1 = 0.0; // initial guess for solution 1
+    double x0_sol2 = 0.0; // initial guess for solution 2 (multi-revolutions only)
+
+    if (this->solverName == "Gooding"){
```

Comment:
:small_red_triangle_down: Let's split out this logic into two methods. Something like a `goodingInitialGuess` and an `izzoInitialGuess`. This pulls the method logic up out of the method which claims to get the guess and to the place where we should be deciding on which guess to get. It also greatly simplifies the cognitive load of this method making it easier to understand and maintain in the future. Finally, it also increases test-ability. While we don't have it now, I am integrating Google Test for direct C++ testing in Basilisk. These methods are exactly methods we need to test, yet don't want to expose to the Python user API. So testing at the C++ level is what we want. Separating these components out will allow us to write tests for each method individually. The test code will also be simpler because of it.